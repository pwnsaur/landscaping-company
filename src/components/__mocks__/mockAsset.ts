import { Asset } from 'contentful';

export const createMockAsset = (): Asset => {
  return {
    sys: {
      id: '1',
      type: 'Asset',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'asset',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'jonathan-kemper-kYl3Yp2hwtQ-unsplash',
      description: '',
      file: {
        url: '//images.ctfassets.net/sandie-clarke-q13Zq1Jufks-unsplash.jpg',
        details: {
          size: 500939,
          image: {
            width: 1920,
            height: 1080,
          },
        },
        fileName: '',
        contentType: 'image/jpeg',
      },
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Asset),
  };
};
