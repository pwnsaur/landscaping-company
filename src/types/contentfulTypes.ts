import type * as CFRichTextTypes from '@contentful/rich-text-types';
import type { Asset, Entry, EntryFields } from 'contentful';

export interface TypeProjectFields {
  title: EntryFields.Symbol;
  slug: EntryFields.Symbol;
  content: CFRichTextTypes.Block | CFRichTextTypes.Inline;
  excerpt?: EntryFields.Symbol;
  coverImage: Asset;
  date: EntryFields.Date;
  images?: Asset[];
}

export type TypeProject = Entry<TypeProjectFields>;

export interface TypeServiceFields {
  title: EntryFields.Symbol;
  slug: EntryFields.Symbol;
  description: CFRichTextTypes.Block | CFRichTextTypes.Inline;
  excerpt?: EntryFields.Symbol;
  coverImage: Asset;
}

export type TypeService = Entry<TypeServiceFields>;

export type FormData = {
  name: string;
  email: string;
  phone: string;
  message: string;
};
