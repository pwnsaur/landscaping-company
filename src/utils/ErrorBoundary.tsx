import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
