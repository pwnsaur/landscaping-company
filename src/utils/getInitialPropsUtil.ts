import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
