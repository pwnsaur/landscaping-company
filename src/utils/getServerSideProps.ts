import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
