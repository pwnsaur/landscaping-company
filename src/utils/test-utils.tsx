import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
