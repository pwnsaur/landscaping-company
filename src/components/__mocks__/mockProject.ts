import { createMockAsset } from '@components/__mocks__/mockAsset';
import { BLOCKS } from '@contentful/rich-text-types';
import { Entry } from 'contentful';

import { TypeProjectFields } from '@/types/contentfulTypes';

export const createMockProject = (): Entry<TypeProjectFields> => {
  return {
    sys: {
      id: '1',
      type: 'Entry',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'project',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'Fifteenth Project',
      slug: 'fifteenth-project',
      content: {
        nodeType: BLOCKS.DOCUMENT,
        data: {},
        content: [],
      },
      coverImage: createMockAsset(),
      date: '2023-04-27',
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Entry<TypeProjectFields>),
    update: () => Promise.resolve({} as Entry<TypeProjectFields>),
  };
};
