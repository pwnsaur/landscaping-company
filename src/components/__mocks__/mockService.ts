import { createMockAsset } from '@components/__mocks__/mockAsset';
import { BLOCKS } from '@contentful/rich-text-types';
import { Entry } from 'contentful';

import { TypeServiceFields } from '@/types/contentfulTypes';

export const createMockService = (): Entry<TypeServiceFields> => {
  return {
    sys: {
      id: '1',
      type: 'Entry',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'service',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'Service',
      slug: 'service',
      excerpt: 'service',
      description: {
        nodeType: BLOCKS.PARAGRAPH,
        data: {},
        content: [],
      },
      coverImage: createMockAsset(),
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Entry<TypeServiceFields>),
    update: () => Promise.resolve({} as Entry<TypeServiceFields>),
  };
};
