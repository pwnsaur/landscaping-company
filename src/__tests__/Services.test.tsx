import { createMockService } from '@/components/__mocks__/mockService';
import Services, { getStaticProps } from '@/pages/services';
import { TypeService } from '@/types/contentfulTypes';
import { render } from '@/utils/test-utils';

const mockService = createMockService();
const mockServices: TypeService[] = [mockService];

describe('Projects page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Services services={mockServices} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should export getStaticProps function', () => {
    expect(getStaticProps).toBeDefined();
  });
});
