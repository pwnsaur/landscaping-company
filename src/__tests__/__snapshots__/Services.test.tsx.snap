// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects page should render correctly 1`] = `
<DocumentFragment>
  .c2 {
  border: none;
  height: 1px;
  background-color: #1a1a1a;
  margin: 1rem 0;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  margin: 0 auto;
  text-align: center;
}

.c7 {
  font-size: clamp(1.2rem, 2vw, 1.5rem);
  font-weight: 400;
  margin: auto;
}

.c5 {
  max-width: 40%;
  height: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c8 {
  margin: 10px 20px;
  font-size: clamp(0.8rem, 1.2vw, 1.2rem);
}

.c6 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 75rem;
  max-width: 48rem;
  padding: 8vh 3vw;
}

.c1 {
  max-width: 25rem;
  text-align: center;
  text-transform: uppercase;
  font-size: clamp(1.2rem, 2vw, 1.5rem);
  font-weight: 400;
  padding-bottom: 3rem;
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      A short description of types of services we offer to our clients
    </h1>
    <hr
      class="c2"
    />
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/services/service"
      >
        <img
          alt="cover-image"
          class="c5"
          data-nimg="1"
          decoding="async"
          height="270"
          src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=1080&q=50"
          srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=640&q=50 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=1080&q=50 2x"
          style="color: transparent;"
          width="480"
        />
        <div
          class="c6"
        >
          <h2
            class="c7"
          >
            Service
          </h2>
          <p
            class="c8"
          >
            service
          </p>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
